### ISSUE TITLE
Can we have the ability to turn off text wrap for new side-by-side diff?

### ISSUE THREAD

**Issue by aybe** [desktop/desktop#11052]
### Describe the feature or problem you’d like to solve

Text wrapping on side-by-side diff can make things more difficult to read in some cases.

### Proposed solution

Allow to disable text-wrap on side by side diffs.

Then of course you'd have to implement synchronized scrollbars for both views.

For the latter, it'd be even great if this could be controlled using the mouse wheel button.

### Additional context

![GitHubDesktop_2020-11-17_17-10-22](https://user-images.githubusercontent.com/1537591/99415137-ce69b780-28f7-11eb-8549-35996bb8cc8c.png)

**Comment by steveward** [comment]
Requested in https://github.com/desktop/desktop/issues/11253:

>Describe the feature or problem you’d like to solve
The change diff viewer currently wraps lines, without an option to disable this functionality. I've never seen a diff tool or any tool dealing with code that wraps lines, but wrapping being present in Desktop makes me realize there are many situations wrapping diffs and code is not ideal.
>
>Proposed solution
If line wrapping were optional, code could remain formatted as intended, readability (for us mortals) would no longer be hindered, differing characters could be compared directly in close spatial context (vs across multiple lines), and more lines could be represented in the view. Even a compromise of only wrapping on whitespace, some selection of stop characters, or something else the grammar of the detected language informs would help readability (comprehension and speed), since users would no longer have to mentally stitch together mid-word breaks.

**Comment by emmadesilva** [comment]
I also would love this. Helps especially on small screens! I was surprised to not see this in the gearbox dropdown menu. 
![image](https://user-images.githubusercontent.com/95144705/209569644-cf9efa69-9bc7-40df-94ab-08979fbfe48f.png)

**Comment by PKPublicCode** [comment]
2.5 years later...

Any updates on this issue? What's conclusion and what's ETA?
Current implementation is not usable.

**Comment by dima954844** [comment]
сб, 17 черв. 2023 р., 14:40 користувач P.K. ***@***.***> пише:

**Comment by MarcMouallem** [comment]
Any update? With everyone on MacBook Airs these days, this is a basic need.

**Comment by dustinlineweber** [comment]
I'd argue that forcing line wrapping is a really bad idea for the Unified Diff view as well.  Especially for code files, since line terminations can have an effect in some languages and definitely (in all languages) formatting is done with **strong** intention.

If anyone is wrapping code for the benefit of viewing, they have done so manually in a well-chosen spot.  Wrapping in a random spot will usually be terrible.

**Comment by vivarose** [comment]
Maybe someone should fork the repo and get this done.

**Comment by turbanisch** [comment]
Yes, please!

**Comment by cagledw** [comment]
weak and worthless!

**Comment by Shawn-sudo** [comment]
@sergiou87 Why "not-planned"?
Many people here would love to see such feature
This feature is absolutely necessary when you're working on a decently big file and the code is indented quite a bit

Check this below image. What is this. Not a good UI I would expect to see on GitHub's official desktop app ☹️
<img width="1171" alt="Image" src="https://github.com/user-attachments/assets/ce0ba2b2-5254-41b3-ba15-8f0c8546a10e" />

**Comment by jkoppin-wave** [comment]
In this most crucial step of the development lifecycle, Azure DevOps is far superior.

**Comment by jibbewillem** [comment]
> [@sergiou87](https://github.com/sergiou87) Why "not-planned"? Many people here would love to see such feature This feature is absolutely necessary when you're working on a decently big file and the code is indented quite a bit
> 
> Check this below image. What is this. Not a good UI I would expect to see on GitHub's official desktop app ☹️ <img alt="Image" width="1171" src="https://private-user-images.githubusercontent.com/60294318/425803988-ce0ba2b2-5254-41b3-ba15-8f0c8546a10e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDU0MjIzMTAsIm5iZiI6MTc0NTQyMjAxMCwicGF0aCI6Ii82MDI5NDMxOC80MjU4MDM5ODgtY2UwYmEyYjItNTI1NC00MWIzLWJhMTUtOGYwYzg1NDZhMTBlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDIzVDE1MjY1MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE5YmJmMzZjOGJiYjAyODlmZTc4YmVkZTYzZTZkMmE0MjlhNWVlYjg0NDk4MzMzYTcyZjFhMjA5MGQ0MGY3MjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.w9WCZUEBGC5kpD5sQiYCJtCQ1B9cP3xVF4T8eOMuu5g">

I had the same issue. Solved it by changing: Settings > Appearance > Diff > Tab Size to `2`. Hope this helps for you as well

**Comment by svenja-haas** [comment]
It looks like this is <i>still</i> not possible? Wtf. I want to use side by side view with diffs since it's better to read, but then line wrap makes it incredibly hard to read again. Why is such an essential feature not planned? This works in VSCode and I am incredibly happy with scrolling sideways for long lines.

**Comment by ZackWaveXR** [comment]
Just opened a ticket and had it closed as a dupe... bringing me here.

I'd like to just bump this thread (6 years later... hope = dead) and throw some weight behind this idea. 

What we're asking for here is not for pure vanity. I legitimately struggle to read newline logs. My coworkers are capable of navigating it just fine but for me it really degrades and slows down my work. In a newly ai-native world we are facing more reviews than ever before so this has moved from a medium impact to a daily blocker for me.

I don't know the science behind it but I know for a fact that I struggle to focus on the deltas when the lines wrap. I have prove this out many times by simply opening up a different editor such as beyond compare (which allows disabling of wrap). My productivity and quality instantly rise and the burden is lifted.

I'm sure someone out there is thinking it so I will just address it... no I don't want to use beyond compare for diffs in my workflow because A) I can't launch an external diff tool from github desktop B) I don't want to have to wait to launch external programs when the diffs are right there INSIDE github desktop C) I love the easy to use interface of GH desktop and have acclimated over years to it for every project I work through

I totally understand how priorities work in software development and wouldn't be surprised to see this skip many more years. I am just trying to present a case for anyone with the same issues I have focusing on multiline differences and anyone who has to deal with the completely flooded ingress of AI generated PRs.

I am suffering and could use what I consider to be a very light lift for quality-of-life. Please add an option to disable wrap in GH desktop and default it to true so the world doesn't even notice. It would mean a great deal to me!

(everyone speak up if you also suffer the same way I do with multiline diffs)

**Comment by cagledw** [comment]
Yes Zack, almost everyone hates wrapping.  And yes, MS knows this.
However, they have decided it isn't worth fixing.

Do what most of us do, find a workflow that doesn't include github diff.

**Comment by ignatremizov** [comment]
I implemented and tested this behavior in desktop-plus/desktop-plus#216.

The patch adds a persisted **Wrap lines** option to Diff Options, supports both unified and split layouts, wraps Markdown-family files at word boundaries, and uses a shared horizontal viewport (including Shift+wheel input) when wrapping is disabled so split panes stay synchronized.

Since this issue is currently marked `not-planned` and Desktop's contribution policy generally reserves external PRs for `help wanted` issues, I am linking the isolated implementation here instead of opening an unsolicited PR against `desktop/desktop`. It includes focused unit coverage and can serve as a reference if the issue is reconsidered.

<img width="539" height="811" alt="Image" src="https://github.com/user-attachments/assets/ce2457cd-a0a7-45cb-8ec4-a143f255389c" />

https://github.com/user-attachments/assets/134c47d5-8945-4a36-a17a-db0fe8dfd734

**Comment by adesai1000** [comment]
Five years is a long time to wait for a checkbox, so I added it to my fork. GlitchHub Desktop has Settings > Appearance > Diff Expansion > "Wrap long lines". Turn it off and long lines scroll horizontally instead of wrapping, in both unified and side-by-side mode. It scales properly with the app's text size setting too.

Not affiliated with GitHub. macOS on Apple silicon only for now, and the build is unsigned, so the first launch is right-click, Open. It uses its own profile folder, so your GitHub Desktop install is untouched.

Download: https://github.com/adesai1000/GlitchHub-Desktop/releases/latest

If you end up using it, please star the repo. That is what helps it reach the other people who upvoted this.
